File: 백준/Bronze/2501. 약수 구하기/약수 구하기.java
import java.util.Scanner;
public class Main{
    public static void main(String[] args){
        Scanner in = new Scanner(System.in);
        int n = in.nextInt(), k = in.nextInt();
        int cnt = 0;
        for(int i = 1; i <= n; i++){
            if(n%i == 0)
                if(++cnt == k){
                    System.out.print(i);
                    break;
                }
        }
        if(cnt != k)
            System.out.print("0");
    }
}

File: 백준/Bronze/2501. 약수 구하기/약수 구하기.kt
import java.util.Scanner
fun main(){
    val sc = Scanner(System.`in`)
    var n = sc.nextInt(); var k = sc.nextInt()
    var cnt = 0;
    for(i in 1..n){
        if(n%i == 0){
            if(++cnt == k){
                print(i)
                break
            }
        }
    }
    if(cnt != k)
    print("0")
}